madmom/audio/filters.py

```diff
@@ -1531,3 +1531,72 @@ def __array_finalize__(self, obj):
         # set default values here
         self.fref = getattr(obj, 'fref', A4)
         self.window = getattr(obj, 'window', self.WINDOW)
+
+
+class SemitoneBandpassFilterbank(object):
+    """
+    Time domain semitone filterbank of elliptic filters as propsed in [1]_.
+
+    Parameters
+    ----------
+    filter_order : int, optional
+        Filter order of elliptic filters.
+    passband_ripple_db : int, optional
+        Passband ripple in [Db]
+    stopband_rejection_db : int, optional
+        Stopband rejection in [Db]
+    q_factor : int, optional
+        Q-factor.
+    sr_from_midi : numpy array, optional
+        Sampling rate for each MIDI note.
+    midi_min : int, optional
+        Lowest MIDI note.
+    midi_max : int, optional
+        Highest MIDI note.
+    fref : float, optional
+        Reference frequency for the first HPCP bin [Hz].
+
+    References
+    ----------
+    .. [1] Meinard Müller,
+            "Information retrieval for music and motion",
+            Berlin: Springer, 2007.
+
+    """
+
+    # use three different sampling rates for the TimeDomainSemitoneFilterbank
+    _samplingrate_for_midi = np.zeros(128, dtype=int)
+    _samplingrate_for_midi[:59] = 22050 / 25
+    _samplingrate_for_midi[59:95] = 22050 / 5
+    _samplingrate_for_midi[95:] = 22050
+
+    def __init__(self, filter_order=8, passband_ripple_db=1,
+                 stopband_rejection_db=50, q_factor=25,
+                 sr_from_midi=_samplingrate_for_midi, midi_min=21,
+                 midi_max=108, fref=A4):
+        self.filter_order = filter_order
+        self.passband_ripple_db = passband_ripple_db
+        self.stopband_rejection_db = stopband_rejection_db
+        self.q_factor = q_factor
+        self.midi_min = midi_min
+        self.midi_max = midi_max
+        self.fref = fref
+        self.bin_frequencies = midi2hz(
+            np.arange(midi_min, midi_max + 1), fref=fref)
+        self.sr_from_midi = sr_from_midi[midi_min - 1:midi_max]
+        self.fmin = self.bin_frequencies[0]
+        self.fmax = self.bin_frequencies[-1]
+        self.num_bins = midi_max - midi_min + 1
```

Comment:
`fmin`, `fmax` and `num_bins` should be properties, not writeable attributes. Maybe these can be inherited from `Filterbank`.